Formula/s/star.rb

```diff
@@ -1,8 +1,9 @@
 class Star < Formula
   desc "Standard tap archiver"
-  homepage "https://cdrtools.sourceforge.net/private/star.html"
-  url "https://downloads.sourceforge.net/project/s-tar/star-1.5.3.tar.bz2"
-  sha256 "070342833ea83104169bf956aa880bcd088e7af7f5b1f8e3d29853b49b1a4f5b"
+  homepage "https://codeberg.org/schilytools/schilytools"
+  url "https://codeberg.org/schilytools/schilytools/archive/2023-09-28.tar.gz"
+  version "1.7.0-2023-09-28"
+  sha256 "564ea2365876a53eba02f184c565016399aee188c26d862589906cf3f92198e6"
   license "CDDL-1.0"
 
```

Comment:
If we want to check the Git tags:

```suggestion

  livecheck do
    url :stable
    regex(/^(\d+(?:[.-]\d+)+)$/i)
    strategy :git do |tags, regex|
      # Find the highest version tag
      tag = tags.map { |tag| tag[regex, 1] }
                .flatten
                .uniq
                .map { |v| Version.new(v) }
                .sort
                .last
      next if tag.blank?

      # Fetch the `star/version.c` file for the highest version tag
      version_file = Homebrew::Livecheck::Strategy.page_content(
        "https://codeberg.org/schilytools/schilytools/raw/tag/#{tag}/star/version.c",
      )
      next if version_file[:content].blank?

      # Identify the `star` version from the `version.c` file
      version = version_file[:content][/strvers\[\]\s*=\s*"([^"]+?)"/i, 1]
      next if version.blank?

      "#{version}-#{tag}"
    end
  end

```

If we want to check release tags:

```suggestion

  livecheck do
    url "https://codeberg.org/schilytools/schilytools/releases"
    regex(%r{href=.*?/src/tag/(\d+(?:[.-]\d+)+)/?["' >]}i)
    strategy :page_match do |page, regex|
      # Find the newest release tag
      tag = page.scan(regex).map { |match| match[0] }&.first
      next if tag.blank?

      # Fetch the `star/version.c` file for the newest release tag
      version_file = Homebrew::Livecheck::Strategy.page_content(
        "https://codeberg.org/schilytools/schilytools/raw/tag/#{tag}/star/version.c",
      )
      next if version_file[:content].blank?

      # Identify the `star` version from the `version.c` file
      version = version_file[:content][/strvers\[\]\s*=\s*"([^"]+?)"/i, 1]
      next if version.blank?

      "#{version}-#{tag}"
    end
  end

```

Reply:
Thanks for helping flush both options out, possibly also pertinent is upstream discussion I raised at https://codeberg.org/schilytools/schilytools/issues/61.

Reply:
If we want to check the Git tags and use the tag that's closest to the date in `version.c`:

```suggestion

  livecheck do
    url :stable
    regex(/^(\d+(?:[.-]\d+)+)$/i)
    strategy :git do |tags, regex|
      # Find the highest version tag
      vtags = tags.map { |tag| tag[regex, 1] }
                  .compact
                  .map { |tag| Version.new(tag) }
      next if vtags.blank?

      # Fetch the `star/version.c` file for the highest version tag
      latest_tag = vtags.max.to_s
      version_file = Homebrew::Livecheck::Strategy.page_content(
        "https://codeberg.org/schilytools/schilytools/raw/tag/#{latest_tag}/star/version.c",
      )
      next if version_file[:content].blank?

      # Identify the `star` version and date from the `version.c` file
      version = version_file[:content][/strvers\[\]\s*=\s*"([^"]+?)"/i, 1]
      date = version_file[:content][/dvers\[\]\s*=\s*"([^"]+?)"/i, 1]
      next if version.blank? || date.blank?

      # Identify the tag that's closest to the version date
      date_version = Version.new(date)
      tag = vtags.select { |v| v >= date_version }.min

      "#{version}-#{tag}"
    end
  end

```

If we want to check release tags and use the tag that's closest to the date in `version.c`:

```suggestion

  livecheck do
    url "https://codeberg.org/schilytools/schilytools/releases"
    regex(%r{href=.*?/src/tag/(\d+(?:[.-]\d+)+)/?["' >]}i)
    strategy :page_match do |page, regex|
      # Find the highest release version tag
      vtags = page.scan(regex).map { |match| match[0] }
                  .map { |tag| Version.new(tag) }
      next if vtags.blank?

      # Fetch the `star/version.c` file for the highest release tag
      latest_tag = vtags.max.to_s
      version_file = Homebrew::Livecheck::Strategy.page_content(
        "https://codeberg.org/schilytools/schilytools/raw/tag/#{latest_tag}/star/version.c",
      )
      next if version_file[:content].blank?

      # Identify the `star` version and date from the `version.c` file
      version = version_file[:content][/strvers\[\]\s*=\s*"([^"]+?)"/i, 1]
      date = version_file[:content][/dvers\[\]\s*=\s*"([^"]+?)"/i, 1]
      next if version.blank? || date.blank?

      # Identify the tag that's closest to the version date
      date_version = Version.new(date)
      tag = vtags.select { |v| v >= date_version }.min

      "#{version}-#{tag}"
    end
  end

```